src/sage/modules/ore_module.py

```diff
@@ -351,26 +369,51 @@ def __classcall_private__(cls, mat, twist, names=None, category=None):
             False
             sage: M2 is M3
             True
+
+        ::
+
+            sage: from sage.modules.ore_module import OreModule
+            sage: mat = matrix(K, [[1, z], [z^2, z^3]])
+            sage: den = z + 1
+            sage: OreModule(mat, S, den)
+            Ore module of rank 2 over Finite Field in z of size 5^3 twisted by z |--> z^5
+            sage: OreModule(mat, S, Factorization([(den, 10)]))
+            Ore module of rank 2 over Finite Field in z of size 5^3 twisted by z |--> z^5
         """
         base = mat.base_ring()
+        if denominator is None:
+            pass
+        elif isinstance(denominator, Factorization):
+            denominator = denominator.base_change(base)
+        else:
+            denominator = Factorization([(base(denominator), 1)])
         if category is None:
             category = OreModules(base, twist)
         rank = mat.nrows()
         if mat.ncols() != rank:
             raise ValueError("matrix must be square")
         names = normalize_names(names, rank)
-        return cls.__classcall__(cls, mat, category._ore, names, category)
+        return cls.__classcall__(cls, mat, category._ore, denominator, names, category)
 
-    def __init__(self, mat, ore, names, category) -> None:
+    def __init__(self, mat, ore, denominator, names, category) -> None:
         r"""
         Initialize this Ore module.
 
         INPUT:
 
-        - ``mat`` -- the matrix defining the action of the Ore variable
+        - ``mat`` -- a matrix; the matrix defining the action of the Ore
+          variable is ``mat``/``denominator``
 
         - ``ore`` -- the underlying Ore polynomial ring
 
+        - ``denominator`` (default: ``None``) -- an instance of
+          :class:`sage.structure.factorization.Factorization`;
+          ``None`` is understood as the empty factorization with
+          value `1`
```

Comment:
```suggestion
        - ``denominator`` -- either ``None`` or an instance of
          :class:`sage.structure.factorization.Factorization`;
          ``None`` is understood as the empty factorization with
          value `1`
```
Here `denominator` is a positional argument